devel/ci/cico.pipeline

```diff
@@ -0,0 +1,158 @@
+/**
+ * This is Bodhi's Jenkins Pipeline Jenkinsfile.
+ *
+ * You can read documentation about this file at https://jenkins.io/doc/book/pipeline/jenkinsfile/.
+ * A useful list of plugins can be found here: https://jenkins.io/doc/pipeline/steps/.
+ */
+
+
+/**
+ * Run the given script on the Duffy node.
+ *
+ * @param script The script to run on the node.
+ */
+def onmyduffynode(String script) {
+    ansiColor('xterm'){
+        timestamps{
+            sh 'ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -l root ${DUFFY_NODE}.ci.centos.org "' + script + '"'
+        }
+    }
+}
+
+
+/**
+ * rsync the given path from the Duffy node back to the control host.
+ *
+ * @param rsyncpath The path to be rsync'd back to the control host.
+ */
+def syncfromduffynode(rsyncpath) {
+    sh 'rsync -e "ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -l root " -Ha --include=' + " ${DUFFY_NODE}.ci.centos.org:~/payload/" + rsyncpath + " ./"
+}
+
+
+/**
+ * rsync the given path from the control host to the Duffy node.
+ *
+ * @param rsyncpath The path to be rsync'd to the Duffy node.
+ */
+def synctoduffynode(rsyncpath) {
+    sh 'rsync -e "ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -l root " -Ha --include=' +  rsyncpath + " ./ ${DUFFY_NODE}.ci.centos.org:~/payload/"
+}
+
+
+/**
+ * Install test dependencies on the Duffy node.
+ */
+def configure_node = {
+    onmyduffynode 'yum -y install epel-release'
+    onmyduffynode 'yum install -y docker python34-click rsync'
+    onmyduffynode 'systemctl start docker'
+}
+
+
+/**
+ * Run the given function up to 5 times (if it fails), with 60 seconds of sleeping in between tries.
+ *
+ * @param f The function to retry.
+ */
+def retry_with_sleep = { f ->
+    try {
+        f()
+    } catch (error) {
+        retry(4) {
+            sleep(60)
+            f()
+        }
+    }
+}
+
+
+/**
+ * Run bodhi-ci on the Duffy node for the given arguments.
+ *
+ * @param release The release to pass to bodhi-ci's -r flag.
+ * @param command The subcommand of bodhi-ci to use.
+ * @param context This parameter is appended to the release and a dash to form the GitHub context.
+ * @param args Any extra arguments to be passed to the command.
+ */
+def bodhi_ci = { String release, String command, String context, String args ->
+    githubNotify context: release + '-' + context, status: 'PENDING'
+
+    try {
+        stage(release + ' ' + command) {
+            timeout(time: 1, unit: 'HOURS') {
+                onmyduffynode 'cd payload && ./devel/ci/bodhi-ci ' + command + ' -r ' + release + ' ' + args
+            }
+        }
+        githubNotify context: release + '-' + context, status: 'SUCCESS'
+    } catch(error) {
+        githubNotify context: release + '-' + context, status: 'FAILURE'
```

Comment:
@bstinsonmhk I am adding one more patch here:

```diff --git a/devel/ci/cico.pipeline b/devel/ci/cico.pipeline
index 9dbd90bd..c7175d6e 100644
--- a/devel/ci/cico.pipeline
+++ b/devel/ci/cico.pipeline
@@ -87,6 +87,7 @@ def bodhi_ci = { String release, String command, String context, String args ->
         githubNotify context: release + '-' + context, status: 'SUCCESS'
     } catch(error) {
         githubNotify context: release + '-' + context, status: 'FAILURE'
+        throw error
     }
 }

```

Without this it still tries to run the tests on builds that never happened, which is not what we want.